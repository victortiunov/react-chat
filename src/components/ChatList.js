import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import List from 'material-ui/List';
import ChatListItem from './ChatListItem';

const styles = () => ({
  chatsList: {
    height: 'calc(100% - 56px)',
    overflowY: 'scroll',
  },
});

const ChatList = ({
  classes, activeChat, chats, setActiveChat, disabled,
}) => (
  <List className={classes.chatsList}>
    {chats.map(chat => (
      <ChatListItem
        /* eslint-disable no-underscore-dangle */
        key={chat._id}
        {...chat}
        isActive={activeChat ? chat._id === activeChat._id : false}
        onClick={() => setActiveChat(chat._id)}
        /* eslint-enable no-underscore-dangle */
        disabled={disabled}
      />
    ))}
  </List>
);

ChatList.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  activeChat: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    creator: PropTypes.shape({
      _id: PropTypes.string.isRequired,
    }).isRequired,
    members: PropTypes.arrayOf(PropTypes.object).isRequired,
  }),
  chats: PropTypes.arrayOf(PropTypes.object).isRequired,
  setActiveChat: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
};

ChatList.defaultProps = {
  activeChat: null,
};

export default withStyles(styles)(ChatList);
