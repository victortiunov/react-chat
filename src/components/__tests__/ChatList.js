/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatList from '../ChatList';

jest.mock('../ChatListItem', () => () => 'ChatListItem');

const mockProps = {
  activeChat: {
    _id: '373737',
  },
  chats: [
    {
      _id: '373737',
      title: 'Gryffindor Talks',
    },
    {
      _id: '737373',
      title: 'Hufflepuff Talks',
    },
    {
      _id: '111111',
      title: 'Ravenclaw Talks',
    },
    {
      _id: '666666',
      title: 'Slytherin Talks',
    },
  ],
  setActiveChat: jest.fn(),
  disabled: false,
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatList {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without active chat', () => {
    const tree = renderer.create(<ChatList {...mockProps} activeChat={null} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled correctly', () => {
    const tree = renderer.create(<ChatList {...mockProps} disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
