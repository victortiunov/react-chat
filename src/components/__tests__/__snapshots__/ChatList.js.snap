// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly 1`] = `
<ul
  className="MuiList-root-7 MuiList-padding-8 ChatList-chatsList-6"
>
  ChatListItem
  ChatListItem
  ChatListItem
  ChatListItem
</ul>
`;

exports[`<ChatList /> renders disabled correctly 1`] = `
<ul
  className="MuiList-root-7 MuiList-padding-8 ChatList-chatsList-6"
>
  ChatListItem
  ChatListItem
  ChatListItem
  ChatListItem
</ul>
`;

exports[`<ChatList /> renders without active chat 1`] = `
<ul
  className="MuiList-root-7 MuiList-padding-8 ChatList-chatsList-6"
>
  ChatListItem
  ChatListItem
  ChatListItem
  ChatListItem
</ul>
`;
